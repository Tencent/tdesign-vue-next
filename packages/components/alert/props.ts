/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * updated at 2021-12-12 19:17:30
 * */

import { TdAlertProps } from './type';
import { PropType } from 'vue';

export default {
  /** 关闭按钮。值为 true 则显示默认关闭按钮；值为 false 则不显示按钮；值类型为 string 则直接显示；值类型为 Function 则可以自定关闭按钮 */
  close: {
    type: [String, Boolean, Function] as PropType<TdAlertProps['close']>,
    default: false,
  },
  /** 内容，同 message */
  default: {
    type: [String, Function] as PropType<TdAlertProps['default']>,
  },
  /** 图标 */
  icon: {
    type: Function as PropType<TdAlertProps['icon']>,
  },
  /** 内容显示最大行数，超出的内容会折叠收起，用户点击后再展开。值为 0 表示不折叠 */
  maxLine: {
    type: Number,
    default: 0,
  },
  /** 内容（子元素） */
  message: {
    type: [String, Function] as PropType<TdAlertProps['message']>,
  },
  /** 跟在告警内容后面的操作区 */
  operation: {
    type: Function as PropType<TdAlertProps['operation']>,
  },
  /** 组件风格 */
  theme: {
    type: String as PropType<TdAlertProps['theme']>,
    default: 'info' as TdAlertProps['theme'],
    validator(val: TdAlertProps['theme']): boolean {
      return ['success', 'info', 'warning', 'error'].includes(val);
    },
  },
  /** 标题 */
  title: {
    type: [String, Function] as PropType<TdAlertProps['title']>,
  },
  /** 关闭按钮点击时触发 */
  onClose: Function as PropType<TdAlertProps['onClose']>,
  /** 告警提示框关闭动画结束后触发 */
  onClosed: Function as PropType<TdAlertProps['onClosed']>,
};
