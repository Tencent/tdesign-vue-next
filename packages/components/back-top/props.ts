/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TdBackTopProps } from './type';
import { PropType } from 'vue';

export default {
  /** 监听滚动的容器。数据类型为 String 时，会被当作选择器处理，进行节点查询。示例：'body' 或 () => document.body */
  container: {
    type: [String, Function] as PropType<TdBackTopProps['container']>,
    default: 'body' as TdBackTopProps['container'],
  },
  /** 回到顶部内容 */
  content: {
    type: [String, Function] as PropType<TdBackTopProps['content']>,
  },
  /** 回到顶部内容，同 `content` */
  default: {
    type: [String, Function] as PropType<TdBackTopProps['default']>,
  },
  /** 回到顶部的耗时单位：毫秒 */
  duration: {
    type: Number,
    default: 200,
  },
  /** 回到顶部相对右下角的位置偏移，示例：[10, 20] 或 ['10em', '8rem'] */
  offset: {
    type: Array as PropType<TdBackTopProps['offset']>,
    default: (): TdBackTopProps['offset'] => ['24px', '80px'],
  },
  /** 回到顶部的形状 */
  shape: {
    type: String as PropType<TdBackTopProps['shape']>,
    default: 'square' as TdBackTopProps['shape'],
    validator(val: TdBackTopProps['shape']): boolean {
      if (!val) return true;
      return ['circle', 'square'].includes(val);
    },
  },
  /** 组件尺寸 */
  size: {
    type: String as PropType<TdBackTopProps['size']>,
    default: 'medium' as TdBackTopProps['size'],
    validator(val: TdBackTopProps['size']): boolean {
      if (!val) return true;
      return ['medium', 'small'].includes(val);
    },
  },
  /** 指定回到该对象。数据类型为 String 时，会被当作选择器处理，进行节点查询。示例：'body' 或 () => document.body */
  target: {
    type: [String, Function] as PropType<TdBackTopProps['target']>,
    default: 'body' as TdBackTopProps['target'],
  },
  /** 组件主题风格，浅色、主色、深色 */
  theme: {
    type: String as PropType<TdBackTopProps['theme']>,
    default: 'light' as TdBackTopProps['theme'],
    validator(val: TdBackTopProps['theme']): boolean {
      if (!val) return true;
      return ['light', 'primary', 'dark'].includes(val);
    },
  },
  /** 滚动高度达到此参数值才出现 */
  visibleHeight: {
    type: [String, Number] as PropType<TdBackTopProps['visibleHeight']>,
    default: '200px' as TdBackTopProps['visibleHeight'],
  },
  /** 点击回到顶部时触发 */
  onClick: Function as PropType<TdBackTopProps['onClick']>,
};
