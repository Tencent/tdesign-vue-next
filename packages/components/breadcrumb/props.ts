/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TdBreadcrumbProps } from './type';
import { PropType } from 'vue';

export default {
  /** 单项最大宽度，超出后会以省略号形式呈现 */
  maxItemWidth: {
    type: String,
    default: undefined,
  },
  /** 面包屑项，功能同 BreadcrumbItem */
  options: {
    type: Array as PropType<TdBreadcrumbProps['options']>,
  },
  /** 自定义分隔符 */
  separator: {
    type: [String, Function] as PropType<TdBreadcrumbProps['separator']>,
  },
  /** 组件风格 */
  theme: {
    type: String as PropType<TdBreadcrumbProps['theme']>,
    default: 'light' as TdBreadcrumbProps['theme'],
    validator(val: TdBreadcrumbProps['theme']): boolean {
      if (!val) return true;
      return ['light'].includes(val);
    },
  },
};
