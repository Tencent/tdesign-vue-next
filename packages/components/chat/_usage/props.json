[
  {
    "name": "animation",
    "type": "enum",
    "defaultValue": "skeleton",
    "options": [
      {
        "label": "skeleton",
        "value": "skeleton"
      },
      {
        "label": "moving",
        "value": "moving"
      },
      {
        "label": "gradient",
        "value": "gradient"
      }
    ]
},
    {
        "name": "layout",
        "type": "enum",
        "defaultValue": "both",
        "options": [
          {
            "label": "both",
            "value": "both"
          },
          {
            "label": "single",
            "value": "single"
          }
        ]
},
{
  "name": "clearHistory",
  "type": "Boolean",
  "defaultValue": true,
  "options": []
},
{
  "name": "textLoading",
  "type": "Boolean",
  "defaultValue": false,
  "options": []
}
]