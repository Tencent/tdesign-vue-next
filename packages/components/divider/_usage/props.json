[
  {
    "name": "align",
    "type": "enum",
    "defaultValue": "center",
    "options": [
      {
        "label": "left",
        "value": "left"
      },
      {
        "label": "right",
        "value": "right"
      },
      {
        "label": "center",
        "value": "center"
      }
    ]
  },
  {
    "name": "dashed",
    "type": "Boolean",
    "defaultValue": false,
    "options": []
  },
  {
    "name": "layout",
    "type": "enum",
    "defaultValue": "horizontal",
    "options": [
      {
        "label": "horizontal",
        "value": "horizontal"
      },
      {
        "label": "vertical",
        "value": "vertical"
      }
    ]
  }
]