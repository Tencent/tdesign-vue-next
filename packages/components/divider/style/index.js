import '../../_common/style/web/components/divider/_index.less';
