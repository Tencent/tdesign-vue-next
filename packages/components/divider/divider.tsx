import { defineComponent } from 'vue';
import props from './props';
import { usePrefixClass } from '../hooks/useConfig';
import { useContent } from '../hooks/tnode';

export default defineComponent({
  name: 'TDivider',

  props: { ...props },

  setup(props) {
    const COMPONENT_NAME = usePrefixClass('divider');
    const renderContent = useContent();
    return () => {
      const { layout, dashed, align } = props;
      const children = renderContent('default', 'content');

      const dividerClassNames = [
        `${COMPONENT_NAME.value}`,
        [`${COMPONENT_NAME.value}--${layout}`],
        {
          [`${COMPONENT_NAME.value}--dashed`]: !!dashed,
          [`${COMPONENT_NAME.value}--with-text`]: !!children,
          [`${COMPONENT_NAME.value}--with-text-${align}`]: !!children,
        },
      ];

      return (
        <div class={dividerClassNames}>
          {children && <span class={`${COMPONENT_NAME.value}__inner-text`}>{children}</span>}
        </div>
      );
    };
  },
});
