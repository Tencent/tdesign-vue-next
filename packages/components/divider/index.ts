import _Divider from './divider';
import { withInstall } from '@tdesign/shared-utils';
import { TdDividerProps } from './type';

import './style';

export * from './type';
export type DividerProps = TdDividerProps;

export const Divider = withInstall(_Divider);
export default Divider;
