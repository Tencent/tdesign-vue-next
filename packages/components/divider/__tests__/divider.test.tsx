import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import { Divider } from '@tdesign/components/divider';
import props from '@tdesign/components/divider/props';

describe('Divider', () => {
  describe('props', () => {
    it(':align[left/right/center]', () => {
      const { validator } = props.align;
      expect(validator('left')).toBeTruthy();
      expect(validator('right')).toBeTruthy();
      expect(validator('center')).toBeTruthy();
      // @ts-expect-error
      expect(validator('top')).toBeFalsy();
      // @ts-expect-error
      expect(validator('')).toBeTruthy();
      expect(validator(undefined)).toBeTruthy();

      // horizontal
      (['left', 'right', 'center'] as const).forEach((align) => {
        const wrapper = mount(<Divider align={align}>Text</Divider>);
        expect(wrapper.classes(`t-divider--with-text-${align}`)).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
      });

      // vertical left
      const wrapper = mount(
        <Divider layout="vertical" align="left">
          text
        </Divider>,
      );
      expect(wrapper.classes('t-divider--with-text-left')).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot('align-vertical-left');
    });

    it(':dashed[true/false]', () => {
      // default
      const wrapper1 = mount(<Divider></Divider>);
      expect(wrapper1.classes('t-divider--dashed')).toBeFalsy();
      expect(wrapper1.element).toMatchSnapshot('dashed-false-default');

      // true
      const wrapper2 = mount(<Divider dashed={true}></Divider>);
      expect(wrapper2.classes('t-divider--dashed')).toBeTruthy();
      expect(wrapper2.element).toMatchSnapshot('dashed-true');

      // false
      const wrapper3 = mount(<Divider dashed={false}></Divider>);
      expect(wrapper3.classes('t-divider--dashed')).toBeFalsy();
      expect(wrapper3.element).toMatchSnapshot('dashed-false-explicit');

      // vertical dashed
      const wrapper4 = mount(<Divider layout="vertical" dashed={true} />);
      expect(wrapper4.classes('t-divider--dashed')).toBeTruthy();
      expect(wrapper4.element).toMatchSnapshot('dashed-vertical-true');
    });

    it(':layout[horizontal/vertical]', () => {
      const { validator } = props.layout;
      expect(validator('horizontal')).toBeTruthy();
      expect(validator('vertical')).toBeTruthy();
      // @ts-expect-error
      expect(validator('other')).toBeFalsy();
      expect(validator(undefined)).toBeTruthy();
      (['horizontal', 'vertical'] as const).forEach((layout) => {
        const wrapper = mount(<Divider layout={layout}></Divider>);
        expect(wrapper.classes(`t-divider--${layout}`)).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    it(':theme[horizontal/vertical]', () => {
      const { validator } = props.theme;
      expect(validator('horizontal')).toBeTruthy();
      expect(validator('vertical')).toBeTruthy();
      // @ts-expect-error types error
      expect(validator('other')).toBeFalsy();
      expect(validator(undefined)).toBeTruthy();
    });

    it(':content[string/function]', () => {
      // string
      const wrapperString = mount(<Divider content="TDesign" />);
      expect(wrapperString.text()).toBe('TDesign');
      expect(wrapperString.element).toMatchSnapshot('content-string');

      // function
      const wrapperTNode = mount(<Divider content={() => <span class="custom-node">TNode</span>} />);
      expect(wrapperTNode.find('.custom-node').exists()).toBeTruthy();
      expect(wrapperTNode.text()).toBe('TNode');
      expect(wrapperTNode.element).toMatchSnapshot('content-TNode');
    });

    it(':content[slot]', () => {
      const wrapperString = mount(<Divider v-slots={{ content: () => 'TDesign' }} />);
      expect(wrapperString.text()).toBe('TDesign');
      expect(wrapperString.element).toMatchSnapshot('content-string');
    });

    it(':default[string/function]', () => {
      // string
      const wrapperString = mount(<Divider default="TDesign" />);
      expect(wrapperString.text()).toBe('TDesign');
      expect(wrapperString.element).toMatchSnapshot('default-string');

      // function
      const wrapperFunction = mount(<Divider v-slots={{ default: () => <span class="custom-node">TNode</span> }} />);
      expect(wrapperFunction.find('.custom-node').exists()).toBeTruthy();
      expect(wrapperFunction.text()).toBe('TNode');
      expect(wrapperFunction.element).toMatchSnapshot();
    });

    it(':default[slot]:should be priority to content', () => {
      const wrapper = mount(<Divider content="prop content" v-slots={{ default: () => 'TDesign' }} />);
      expect(wrapper.text()).toBe('TDesign');
      expect(wrapper.element).toMatchSnapshot('default-priority');
    });
  });
});
