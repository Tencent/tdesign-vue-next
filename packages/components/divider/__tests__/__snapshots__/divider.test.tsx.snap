// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > props > :align[left/right/center] > align-vertical-left 1`] = `
<div
  class="t-divider t-divider--vertical t-divider--with-text t-divider--with-text-left"
>
  <span
    class="t-divider__inner-text"
  >
    
    text
    
  </span>
</div>
`;

exports[`Divider > props > :align[left/right/center] 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-left"
>
  <span
    class="t-divider__inner-text"
  >
    
    Text
    
  </span>
</div>
`;

exports[`Divider > props > :align[left/right/center] 2`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-right"
>
  <span
    class="t-divider__inner-text"
  >
    
    Text
    
  </span>
</div>
`;

exports[`Divider > props > :align[left/right/center] 3`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    
    Text
    
  </span>
</div>
`;

exports[`Divider > props > :content[string/function] > content-TNode 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    <span
      class="custom-node"
    >
      TNode
    </span>
  </span>
</div>
`;

exports[`Divider > props > :content[string/function] > content-string 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    TDesign
  </span>
</div>
`;

exports[`Divider > props > :dashed[true/false] > dashed-false-default 1`] = `
<div
  class="t-divider t-divider--horizontal"
>
  <!---->
</div>
`;

exports[`Divider > props > :dashed[true/false] > dashed-false-explicit 1`] = `
<div
  class="t-divider t-divider--horizontal"
>
  <!---->
</div>
`;

exports[`Divider > props > :dashed[true/false] > dashed-true 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--dashed"
>
  <!---->
</div>
`;

exports[`Divider > props > :dashed[true/false] > dashed-vertical-true 1`] = `
<div
  class="t-divider t-divider--vertical t-divider--dashed"
>
  <!---->
</div>
`;

exports[`Divider > props > :default[string/function] > default-string 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    TDesign
  </span>
</div>
`;

exports[`Divider > props > :default[string/function] 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    
    <span
      class="custom-node"
    >
      TNode
    </span>
    
  </span>
</div>
`;

exports[`Divider > props > :layout[horizontal/vertical] 1`] = `
<div
  class="t-divider t-divider--horizontal"
>
  <!---->
</div>
`;

exports[`Divider > props > :layout[horizontal/vertical] 2`] = `
<div
  class="t-divider t-divider--vertical"
>
  <!---->
</div>
`;

exports[`Divider > props > default should be priority to content > default-priority 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    
    TDesign
    
  </span>
</div>
`;
